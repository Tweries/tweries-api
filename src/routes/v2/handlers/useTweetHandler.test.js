const fetch = require('node-fetch');
const makeT = require('../../handlers/makeT');
const {
  matcherV2,
  mockSend,
  reqBase,
  resBase,
  tweet
} = require('../../handlers/mocks');
const useTweetHandler = require('./useTweetHandler');

jest.mock('../../handlers/makeT');

const req = {
  ...reqBase,
  body: {
    tweetUrl: 'https://twitter.com/musk_china/status/1199474666412236800',
    userId: 'twitter|1183836409850814464'
  }
};

const res = { ...resBase };

describe('useTweetHandler', () => {
  beforeEach(() => {
    fetch.mockResponses(
      [
        JSON.stringify({ access_token: 'mock_access_token' }),
        { status: '200' }
      ],
      [
        JSON.stringify({
          identities: [
            {
              access_token: 'mock_access_token',
              access_token_secret: 'mock_access_token_secret',
              provider: 'twitter'
            }
          ]
        }),
        { status: 200 }
      ]
    );

    mockSend.mockReset();
  });

  it('response w/o error', async () => {
    makeT.mockImplementation(() => ({ get: jest.fn(() => ({ data: tweet })) }));

    await useTweetHandler(req, res);

    expect(mockSend.mock.calls[0][0]).toMatchSnapshot(matcherV2);
  });

  it('response w/ error', async () => {
    makeT.mockImplementation(() => ({
      get: jest.fn(() => {
        throw new Error('Oh Noes!');
      })
    }));

    await useTweetHandler(req, res);

    expect(mockSend.mock.calls[0][0]).toMatchSnapshot(matcherV2);
  });
});
