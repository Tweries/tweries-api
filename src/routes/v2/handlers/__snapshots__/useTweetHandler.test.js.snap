// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useTweetHandler response w/ error 1`] = `
Object {
  "data": undefined,
  "error": Object {},
  "meta": Object {
    "env": "test",
    "name": "tweries-api",
    "timestamp": Any<Number>,
    "version": Any<String>,
  },
}
`;

exports[`useTweetHandler response w/o error 1`] = `
Object {
  "data": Object {
    "contributors": null,
    "coordinates": null,
    "created_at": "Tue Nov 26 23:47:00 +0000 2019",
    "entities": Object {
      "hashtags": Array [],
      "symbols": Array [],
      "urls": Array [],
      "user_mentions": Array [],
    },
    "favorite_count": 0,
    "favorited": false,
    "geo": null,
    "id": 1199474666412236800,
    "id_str": "1199474666412236800",
    "in_reply_to_screen_name": null,
    "in_reply_to_status_id": null,
    "in_reply_to_status_id_str": null,
    "in_reply_to_user_id": null,
    "in_reply_to_user_id_str": null,
    "is_quote_status": false,
    "lang": "en",
    "place": null,
    "retweet_count": 0,
    "retweeted": false,
    "source": "<a href=\\"https://tweries.com\\" rel=\\"nofollow\\">TweriesApp</a>",
    "text": "test",
    "truncated": false,
    "user": Object {
      "can_media_tag": true,
      "contributors_enabled": false,
      "created_at": "Mon Oct 14 20:06:48 +0000 2019",
      "default_profile": true,
      "default_profile_image": true,
      "description": "",
      "entities": Object {
        "description": Object {
          "urls": Array [],
        },
      },
      "favourites_count": 3,
      "follow_request_sent": false,
      "followed_by": false,
      "followers_count": 0,
      "following": false,
      "friends_count": 2,
      "geo_enabled": false,
      "has_extended_profile": false,
      "id": 1183836409850814500,
      "id_str": "1183836409850814464",
      "is_translation_enabled": false,
      "is_translator": false,
      "lang": null,
      "listed_count": 0,
      "location": "",
      "name": "china-musk",
      "notifications": false,
      "profile_background_color": "F5F8FA",
      "profile_background_image_url": null,
      "profile_background_image_url_https": null,
      "profile_background_tile": false,
      "profile_image_url": "http://abs.twimg.com/sticky/default_profile_images/default_profile_normal.png",
      "profile_image_url_https": "https://abs.twimg.com/sticky/default_profile_images/default_profile_normal.png",
      "profile_link_color": "1DA1F2",
      "profile_sidebar_border_color": "C0DEED",
      "profile_sidebar_fill_color": "DDEEF6",
      "profile_text_color": "333333",
      "profile_use_background_image": true,
      "protected": false,
      "screen_name": "musk_china",
      "statuses_count": 56,
      "time_zone": null,
      "translator_type": "none",
      "url": null,
      "utc_offset": null,
      "verified": false,
    },
  },
  "error": undefined,
  "meta": Object {
    "env": "test",
    "name": "tweries-api",
    "timestamp": Any<Number>,
    "version": Any<String>,
  },
}
`;
